.github/workflows/ci.yml

```diff
@@ -27,7 +27,7 @@ jobs:
     - name: Install Rust toolchain
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: nightly
+        toolchain: nightly-2023-06-15
```

Comment:
I don't love this change, but I don't have a better proposal.  Please put a comment block here, with the same explanation you wrote in the PR comment, so we don't lose track of why this very specific nightly version is pinned here.